Userland/Libraries/LibJS/Runtime/RegExpPrototype.cpp

```diff
@@ -281,4 +282,119 @@ JS_DEFINE_NATIVE_FUNCTION(RegExpPrototype::symbol_match)
     return vm.call(*exec, rx, js_string(vm, s));
 }
 
+JS_DEFINE_NATIVE_FUNCTION(RegExpPrototype::symbol_replace)
+{
+    auto string_value = vm.argument(0);
+    auto replace_value = vm.argument(1);
+
+    // https://tc39.es/ecma262/#sec-regexp.prototype-@@replace
+    auto rx = regexp_object_from(vm, global_object);
+    if (!rx)
+        return {};
+    auto string = string_value.to_string(global_object);
+    if (vm.exception())
+        return {};
+
+    auto global_value = rx->get(vm.names.global).value_or(js_undefined());
+    if (vm.exception())
+        return {};
+
+    bool global = global_value.to_boolean();
+    if (global)
+        rx->regex().start_offset = 0;
+
+    // FIXME: Implement and use RegExpExec - https://tc39.es/ecma262/#sec-regexpexec
+    auto* exec = get_method(global_object, rx, vm.names.exec);
+    if (!exec)
+        return {};
+
+    Vector<Value> results;
+
+    while (true) {
+        auto result = vm.call(*exec, rx, string_value);
+        if (result.is_null()) {
+            break;
+        }
+
+        results.append(move(result));
+        if (!global) {
+            break;
+        }
+
+        String match_str = results.last().as_array().get(0).to_string(global_object);
```

Comment:
This blows up when messing with the RegExp object's `exec` method, e.g.:

```js
> r = /foo/
[RegExp] /foo/
> r.exec = () => 42
[ScriptFunction] exec
> r[Symbol.replace]()
js: ../../Userland/Libraries/LibJS/Runtime/Value.cpp:225: JS::Array& JS::Value::as_array(): Assertion `is_array()' failed.
```

spec says `Let matchStr be ? ToString(? Get(result, "0")).`, which works with any object (but depends on `RegExpExec` throwing for type other than object or null).